### PR TITLE
Update release issue template

### PR DESCRIPTION
**Pull request by zelinh**
Signed-off-by: Zelin Hao <zelinhao@amazon.com>

### Description
Update release issue template based on the 2.4.0 release issue https://github.com/opensearch-project/opensearch-build/issues/2649.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 1]
```diff
@@ -42,17 +42,22 @@ To aid in understanding the state of the release there is a table with status in
...
-### CI/CD (Feature Freeze) - _Ends __REPLACE_RELEASE-minus-14-days__
+### Campaigns
 
-- [ ] Create Jenkins workflows that run daily snapshot builds for OpenSearch and OpenSearch Dashboards. 
-- [ ] Increment each component version to {{ env.VERSION }} and ensure working CI in component repositories.
-- [ ] Make pull requests to add each component to [manifests/{{ env.VERSION }}/opensearch-{{ env.VERSION }}.yml](/opensearch-project/opensearch-build/tree/main/manifests/{{ env.VERSION }}/opensearch-{{ env.VERSION }}.yml) and [manifests/{{ env.VERSION }}/opensearch-dashboards-{{ env.VERSION }}.yml](/opensearch-project/opensearch-build/tree/main/manifests/{{ env.VERSION }}/opensearch-dashboards-{{ env.VERSION }}.yml) with the corresponding checks.
-- [ ] OpenSearch / OpenSearch-Dashboards core and components teams finalize their features
+__REPLACE with OpenSearch wide initiatives to improve quality and consistency.__
+
+### Release branch readiness - _Ends __REPLACE_RELEASE-minus-14-days__
```
Comment (dblock) [inline .github/ISSUE_TEMPLATE/release_template.md:49]: Let's Capitalize These Titles Consistently?

I think this should just say "Release Branch"

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 2]
```diff
@@ -42,17 +42,22 @@ To aid in understanding the state of the release there is a table with status in
...
+- [ ] Plugins team to ensure working CI in component repositories and merge the version increment PRs.
 - [ ] OpenSearch / OpenSearch-Dashboards core cut branch `<MajorVersion>.<MinorVersion>` early.
+- [ ] All component repos create `<MajorVersion>.<MinorVersion>` branch for the release.
 
-### Campaigns
+### CI/CD (Feature Freeze) - _Ends __REPLACE_RELEASE-minus-12-days__
 
-__REPLACE with OpenSearch wide initiatives to improve quality and consistency.__
+- [ ] OpenSearch / OpenSearch-Dashboards core and components teams finalize their features
+- [ ] Create Jenkins workflows that run daily snapshot builds for OpenSearch and OpenSearch Dashboards.
```
Comment (dblock) [inline .github/ISSUE_TEMPLATE/release_template.md:59]: Link to an existing workflow/PR/Jenkins?

File: .github/ISSUE_TEMPLATE/release_template.md (hunk 1 above)
Comment (zelinh) [reply .github/ISSUE_TEMPLATE/release_template.md]: Thanks @dblock. Updated all of following suggestions in latest commit.

**Review by dblock — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 3]
```diff
@@ -42,17 +42,22 @@ To aid in understanding the state of the release there is a table with status in
...
 
-- [ ] Create Jenkins workflows that run daily snapshot builds for OpenSearch and OpenSearch Dashboards. 
-- [ ] Increment each component version to {{ env.VERSION }} and ensure working CI in component repositories.
-- [ ] Make pull requests to add each component to [manifests/{{ env.VERSION }}/opensearch-{{ env.VERSION }}.yml](/opensearch-project/opensearch-build/tree/main/manifests/{{ env.VERSION }}/opensearch-{{ env.VERSION }}.yml) and [manifests/{{ env.VERSION }}/opensearch-dashboards-{{ env.VERSION }}.yml](/opensearch-project/opensearch-build/tree/main/manifests/{{ env.VERSION }}/opensearch-dashboards-{{ env.VERSION }}.yml) with the corresponding checks.
-- [ ] OpenSearch / OpenSearch-Dashboards core and components teams finalize their features
+__REPLACE with OpenSearch wide initiatives to improve quality and consistency.__
+
+### Release Branch - _Ends __REPLACE_RELEASE-minus-14-days__
+
+- [ ] Component versions are auto-incremented to {{ env.VERSION }} version
```
Comment (dblock) [inline .github/ISSUE_TEMPLATE/release_template.md:51]: Missing period ;)

Comment (zelinh) [reply .github/ISSUE_TEMPLATE/release_template.md]: Thanks! Updated few others as well.

**Review by dblock — APPROVED** [review]

**Pull request merged**